src/middlewares/condom.rs

```diff
@@ -0,0 +1,139 @@
+/// An entry-point protection middleware that prevents malicious WASM files from reaching the compilation step.
+use std::fmt;
+use wasmer::{
+    FunctionMiddleware, LocalFunctionIndex, MiddlewareError, MiddlewareReaderState,
+    ModuleMiddleware,
+};
+use wasmer_types::ModuleInfo;
+
+/// The module-level export limit middleware, named `CondomMiddleware`.
+pub struct CondomMiddleware {
+    /// Maximum allowed number of exports.
+    max_exports: Option<usize>,
+}
+
+impl CondomMiddleware {
+    /// Creates a new `CondomMiddleware`.
+    pub fn new(max_exports: Option<usize>) -> Self {
+        Self { max_exports }
+    }
+}
+
+impl fmt::Debug for CondomMiddleware {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("CondomMiddleware")
+            .field("max_exports", &self.max_exports)
+            .finish()
+    }
+}
+
+impl ModuleMiddleware for CondomMiddleware {
+    /// This method is called before applying middleware on functions,
+    /// and it will enforce the export limits on the module.
+    /*fn transform_module_info(&self, module_info: &mut ModuleInfo) -> Result<(), MiddlewareError> {
```

Comment:
why is this commented  ?

Reply:
wanted to make it build to be able to `cargo check` and work on massa side

Reply:
got it, let's just make sure we don't forget to uncomment

Reply:
Done as https://github.com/massalabs/massa/pull/4741 is building :) (no versioning for now)